### PR TITLE
connect timeout for session

### PR DESCRIPTION
**Pull request by phact**
Helps for connecting from far away / slow connections

### CODE REVIEW

File: proxy/run.go [hunk 1]
```diff
@@ -184,6 +184,7 @@ func Run(ctx context.Context, args []string) int {
 		Tokens:            cfg.Tokens,
 		Peers:             cfg.Peers,
 		IdempotentGraph:   cfg.IdempotentGraph,
+		ConnectTimeout:    cfg.AstraTimeout,
```
Comment (absurdfarce) [inline proxy/run.go:187]: This feels like a mixing of concerns.  AstraTimeout is currently used to specify timeouts for metadata and SCB retrieval, both of which are very Astra-specific.  Turning it into a general connect timeout feels like a fairly significant change to it's meaning.

How about if instead we introduce an explicit connect timeout CLI param?  We can then set ConnectTimeout throughout the code base via something like:

--connect-timeout CLI param > --astra-timeout CLI param > default value

Winds up accomplishing very much the same thing while still maintaining the separation of concerns.

Thoughts?

Comment (absurdfarce) [inline proxy/run.go:187]: These feel like two different kinds of timeouts.  AstraTimeout currently governs the timeout for actually retrieving metadata or loading the bundle from Astra, both of which are very explicitly related to Astra.  Changing this to also be a general connect timeout feels like a mixing of concerns.

I very much like the idea of being able to explicitly specify a connect timeout, however.  Maybe we add that as an explicit CLI flag, perhaps using Astra timeout if that's specified but the connect timeout isn't?  i.e. connect timeout = connect timeout || astra timeout || some default value?

Comment (absurdfarce) [reply proxy/run.go:187]: A note to future implementors (quite possibly me): there's also an interesting question here about whether we want to make these connect timeouts more granular.  We have potentially distinct timeouts for establishing the control connection (ConnectCluster()) and the connection pools to the various nodes (ConnectSession()).  We may want to enable users to support a longer control connection timeout while enabling more restrictive timeouts for the connection pools.

I'm not :100: convinced this is useful... more something to mull over.

File: proxy/proxy.go [hunk 2]
```diff
@@ -172,6 +173,7 @@ func (p *Proxy) Connect() error {
 		HeartBeatInterval: p.config.HeartBeatInterval,
 		IdleTimeout:       p.config.IdleTimeout,
 		Logger:            p.logger,
+		ConnectTimeout:    p.config.ConnectTimeout,
```
Comment (joao-r-reis) [inline proxy/proxy.go:176]: So this `ConnectTimeout` already existed but it was not being populated anywhere?

Comment (absurdfarce) [reply proxy/proxy.go:176]: Yeah, it's used in cluster and session setup but isn't populated directly anywhere from CLI flags @joao-r-reis

**Comment by absurdfarce** [comment]
This has been resolved by #117.  Closing this as a dupe, please re-open if there's something we missed in the other impl.

**Pull request closed without merging**